In [6]:
cd project_3/

/home/jovyan/work/dsi-plus-2/Projects/project_3


In [10]:
run __init.py__

### importing training and target data

In [12]:
DB_trimmed_df3.shape

(6506, 20)

In [13]:
DB_target.shape

(6506,)

In [15]:
DB_target.head()

0    1
1    1
2    1
3    1
4    0
Name: target, dtype: int64

In [14]:
DB_trimmed_df3.head()

feat_639  feat_956  feat_341  feat_315  feat_867  feat_269  feat_336  \
0  0.660916  1.088752  1.188703 -0.515358  0.955454  1.101862 -0.075655   
1 -3.165634 -2.523248 -5.627815  0.661298  4.071490 -1.842153 -5.979303   
2 -0.343109 -0.621201 -2.602240 -0.549913 -0.136897 -3.181469 -1.688440   
3  1.353230  0.833855  0.134574 -0.190346 -0.852043 -1.593220  1.094138   
4 -3.097087 -2.029956 -6.348887 -1.210703  1.537945 -2.622201 -3.810353   

   feat_395  feat_701  feat_724  feat_257  feat_526  feat_769  feat_829  \
0 -1.403369 -1.120387  0.477059  1.538805  0.110727 -0.311059  2.449488   
1 -4.490868 -2.609116  2.173005 -0.594964  3.090302  4.475702 -1.123622   
2 -0.219289  2.005632  0.186641 -1.204416 -0.236597  1.644917 -2.236824   
3  0.395046  0.947323  0.447420  0.610367  0.130765  1.223934  1.354489   
4 -5.264516 -0.340054  4.948624  0.610158  2.813066  3.107586 -2.873155   

   feat_920  feat_736  feat_808  feat_504  feat_681  feat_308  
0 -2.098730 -1.308329 -0.473198 -0.056173  1.334591  2.122377  
1 -1.925724 -4.241024  1.926215 -0.142134 -1.204221  1.903192  
2  1.470112 -0.885454  1.530122  1.143919 -2.159218 -0.487422  
3  0.095909  0.823544 -0.200368  1.967008 -2.256630 -1.562784  
4  1.047722 -3.454296 -0.341100 -3.458921  0.777129  0.920890

### Setting up Logistic Regression Model

In [16]:
# Train-test split the data
X_train, X_test, y_train, y_test = train_test_split(DB_trimmed_df3, DB_target, random_state = 42)


In [18]:
# Instantiate your pipeline
logreg_pipeline = Pipeline([
    ('scaler', StandardScaler()),
    ('regr', LogisticRegression(),
    )
])

In [21]:
# Fit the pipeline to your training features and target
logreg_pipeline.fit(X_train, y_train)  ###acale and fit all models you choose at once!!!!
# logreg_pipeline.transform(X_test)

Pipeline(memory=None,
     steps=[('scaler', StandardScaler(copy=True, with_mean=True, with_std=True)), ('regr', LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
          intercept_scaling=1, max_iter=100, multi_class='ovr', n_jobs=1,
          penalty='l2', random_state=None, solver='liblinear', tol=0.0001,
          verbose=0, warm_start=False))])

In [22]:
logreg_validation_score = logreg_pipeline.score(X_test, y_test)  
logreg_validation_score

0.59496004917025203

### Setting up KNN Classifier Model

In [23]:


# Instantiate  pipeline.  No hyperparameters
KNN_pipeline = Pipeline([
    ('scaler', StandardScaler()),
    ('regr', KNeighborsClassifier(),
    )
])



In [24]:
# Fit the pipeline to your training features and target
KNN_pipeline.fit(X_train, y_train)  ###scale and fit all models you choose at once!!!!
# KNN_pipeline.transform(X_test)

Pipeline(memory=None,
     steps=[('scaler', StandardScaler(copy=True, with_mean=True, with_std=True)), ('regr', KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
           metric_params=None, n_jobs=1, n_neighbors=5, p=2,
           weights='uniform'))])

In [25]:
KNN_validation_score = KNN_pipeline.score(X_test, y_test)  
KNN_validation_score

0.79287031346035652

###  Setting up Decision Tree Classifier model¶


In [26]:
DTR_pipeline = Pipeline([ # Instantiate  pipeline.  No hyperparameters

    ('scaler', StandardScaler()),
    ('regr', DecisionTreeClassifier(),
    )
])

In [27]:
DTR_pipeline.fit(X_train, y_train)

Pipeline(memory=None,
     steps=[('scaler', StandardScaler(copy=True, with_mean=True, with_std=True)), ('regr', DecisionTreeClassifier(class_weight=None, criterion='gini', max_depth=None,
            max_features=None, max_leaf_nodes=None,
            min_impurity_decrease=0.0, min_impurity_split=None,
            min_samples_leaf=1, min_samples_split=2,
            min_weight_fraction_leaf=0.0, presort=False, random_state=None,
            splitter='best'))])

In [28]:
DTR_validation_score = DTR_pipeline.score(X_test, y_test)
DTR_validation_score


0.71542716656422867

### Setting up SVC pipeline

In [29]:
SVC_pipeline = Pipeline([ # Instantiate  pipeline.  No hyperparameters

    ('scaler', StandardScaler()),
    ('regr', SVC(),
    )
])


In [30]:
SVC_pipeline.fit(X_train, y_train)

Pipeline(memory=None,
     steps=[('scaler', StandardScaler(copy=True, with_mean=True, with_std=True)), ('regr', SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
  decision_function_shape='ovr', degree=3, gamma='auto', kernel='rbf',
  max_iter=-1, probability=False, random_state=None, shrinking=True,
  tol=0.001, verbose=False))])

In [31]:
SVC_validation_score = SVC_pipeline.score(X_test, y_test)
SVC_validation_score

0.75599262446220039

### Comparison of benchmarking to Validating.

### Preliminary benchmarking accuracy scores (before feature reduction, no hyperparameter tuning):


 - Logistic Regression Benchmark : 0.50
 - KNN Classifier Benchmark : 0.49 
 - Decision Tree Classifier Benchmark : 0.50
 - Support Vector Classifier Benchmark : 0.49

### Final accuracy scores (after feature reduction, no hyperparameter tuning):


- Logistic Reression : 0.59
- KNN Classifier : 0.79
- DecisionTree Classifier : 0.72
- SVC Classifier : 0.75